### PR TITLE
Fixed authentication

### PR DESCRIPTION
**Pull request by dbyington**

### CODE REVIEW

File: middlewares/authentication.js [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
-const _getAccessToken = async (ctx) => {
+async function _getAccessToken (ctx) {
+
   const form = {
     client_id: process.env.CLIENT_ID,
     client_secret: process.env.CLIENT_SECRET,
     grant_type: 'authorization_code',
-    redirect_uri: 'https://me/authorize',
-    code: ctx.params.code
+    redirect_uri: 'http://localhost:3000/authorize',
```
Comment (arol) [inline middlewares/authentication.js]: Use `process.env`

**Review by arol — CHANGES_REQUESTED** [review]
I'll make the changes.

File: middlewares/authentication.js [hunk 2]
```diff
@@ -51,33 +51,30 @@ const _getAccessToken = async (ctx) => {
...
     ctx.throw(401, 'unauthorized');
   }
-  // made it through the gaunlet, check the access_token
-  const accessToken = ctx.header.authorization.split(' ')[1];
-  const user = User.find({access_token: accessToken});
-  if (!user['access_token']) {
-    // not a valid access_token
-    ctx.throw(401, 'unauthorized');
+  if (!ctx.state.accessToken) {
+    const user = User.find({access_token: accessToken});
```
Comment (arol) [inline middlewares/authentication.js]: `accessToken` is not defined.

**Review by arol — CHANGES_REQUESTED** [review]

File: middlewares/authentication.js (hunk 2 above)
Comment (arol) [reply middlewares/authentication.js]: I solved some linter errors, but I don't know how to solve this one? Did you want to take the `accessToken` from the headers?

**Comment by dbyington** [comment]
Fixed the check for access_token by awaiting the call to fineOne().

**Review by arol — APPROVED** [review]

**Pull request merged**